### ISSUE TITLE
Bug in jnp.take with negative indexing

### ISSUE THREAD

**Issue by katebaumli** [google/jax#5766]
Hello, I am seeing the following issue with jax.numpy.take and -1 index: 

`A = jnp.array([[1, 2], [3, 4], [5, 6]])`
`A[0]` --> [1, 2]
`A[-1]` --> [5, 6]
`A.take([0, -1], axis=0)` --> [[1, 2], [1, 2]]
`jnp.take(A, [0, -1], axis=0)` --> [[1, 2], [1, 2]]

Thanks!

**Comment by jakevdp** [comment]
Thanks for the report! It looks like the issue is this is lowered to `lax.gather`, which doesn't support negative indices.

One workaround currently would be to use `mode='wrap'`, which does handle negative indices correctly.
